### ISSUE TITLE
expl_impl_clone_on_copy false positive with repr(packed) structs

### ISSUE THREAD

**Issue by TDplai** [rust-lang/rust-clippy#10188]
### Summary

The expl_impl_clone_on_copy lint fires on `#[repr(packed)]` structs with type parameters and suggests adding the `#[derive(Clone)]` annotation instead. Upon applying the suggestion, the compiler produces an error:

```
error: `Clone` can't be derived on this `#[repr(packed)]` struct with type or const parameters
 --> src/lib.rs:4:16
  |
4 | #[derive(Copy, Clone)]
  |                ^^^^^
  |
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #82523 <https://github.com/rust-lang/rust/issues/82523>
  = note: `#[deny(unaligned_references)]` on by default
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Lint Name

expl_impl_clone_on_copy

### Reproducer

I tried this code:

```rust
#![warn(clippy::expl_impl_clone_on_copy)]

#[repr(packed)]
#[derive(Copy)]
struct Packed<T>(T);

impl<T: Copy> Clone for Packed<T> {
    fn clone(&self) -> Self {
        *self
    }
}
```

I saw this happen:

```
warning: you are implementing `Clone` explicitly on a `Copy` type
  --> src/lib.rs:7:1
   |
7  | / impl<T: Copy> Clone for Packed<T> {
8  | |     fn clone(&self) -> Self {
9  | |         *self
10 | |     }
11 | | }
   | |_^
   |
note: consider deriving `Clone` or removing `Copy`
  --> src/lib.rs:7:1
   |
7  | / impl<T: Copy> Clone for Packed<T> {
8  | |     fn clone(&self) -> Self {
9  | |         *self
10 | |     }
11 | | }
   | |_^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expl_impl_clone_on_copy
note: the lint level is defined here
  --> src/lib.rs:1:9
   |
1  | #![warn(clippy::expl_impl_clone_on_copy)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen:

Lint does not fire

### Version

```text
rustc 1.66.1 (90743e729 2023-01-10)
binary: rustc
commit-hash: 90743e7298aca107ddaa0c202a4d3604e29bfeb6
commit-date: 2023-01-10
host: x86_64-unknown-linux-gnu
release: 1.66.1
LLVM version: 15.0.2
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
